state-sync/diem-data-client/src/diemnet/metrics.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use diem_crypto::_once_cell::sync::Lazy;
+use diem_metrics::{
+    register_histogram_vec, register_int_counter_vec, HistogramTimer, HistogramVec, IntCounterVec,
+};
+
+pub const TOTAL: &str = "TOTAL";
+
+/// Counter for tracking sent requests
+pub static SENT_REQUESTS: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_data_client_sent_requests",
+        "Counters related to sent requests",
+        &["request_types"]
+    )
+    .unwrap()
+});
+
+/// Counter for tracking success responses
+pub static SUCCESS_RESPONSES: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_data_client_success_responses",
+        "Counters related to success responses",
+        &["response_type"]
+    )
+    .unwrap()
+});
+
+/// Counter for tracking error responses
+pub static ERROR_RESPONSES: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_data_client_error_responses",
+        "Counters related to error responses",
+        &["response_type"]
+    )
+    .unwrap()
+});
+
+/// Counter for tracking peer selection errors
+pub static PEER_SELECTION_ERRORS: Lazy<IntCounterVec> = Lazy::new(|| {
```

Comment:
I'm separating `PeerSelectionError` from the errors related to the received responses. If you want to drop this or if you want to count `PeerSelectionError` and response errors together, please let me know.

Reply:
To be honest, I don't think we need this for now. If we get a peer selection error, we'll only be able to debug it using the logs. Moreover, we won't be sending any requests or receiving responses, so we'll also know that something is wrong. Perhaps let's remove this until we can see a convincing reason for it?

Reply:
Removed the counter that tracks `PeerSelectionError`.

Yeah I agree that tracking the number of `PeerSelectionError` errors does not seem necessary/useful.